mysql-test/suite/innodb/t/percona_fpt_tablespace_encrypt_2.test

```diff
@@ -0,0 +1,464 @@
+--source include/wait_condition.inc
+
+# Make sure ts file is updated with new records in table
+set global innodb_buf_flush_list_now = 1;
+
+--echo # After restart/recovery, check that Encryption was roll-forward
+SELECT NAME, ENCRYPTION FROM INFORMATION_SCHEMA.INNODB_TABLESPACES WHERE NAME='test/t1';
+SELECT * FROM t1 LIMIT 10;
+
+--echo #########################################################################
+--echo # RESTART 2 : WITHOUT KEYRING PLUGIN
+--echo #########################################################################
+let $restart_parameters = restart:;
+--source include/restart_mysqld_no_echo.inc
+
+# We shouldn't be able to read t1 records as it is encrypted now and keyring
+# plugin is not loaded.
+--error ER_CANNOT_FIND_KEY_IN_KEYRING
+SELECT * FROM t1 LIMIT 10;
+
+--echo #########################################################################
+--echo # RESTART 3 : WITH KEYRING PLUGIN
+--echo #########################################################################
+let $restart_parameters = restart: --early-plugin-load=keyring_file=$KEYRING_PLUGIN --loose-keyring_file_data=$MYSQL_TMP_DIR/mysecret_keyring $KEYRING_PLUGIN_OPT;
+--source include/restart_mysqld_no_echo.inc
+
+SELECT NAME, ENCRYPTION FROM INFORMATION_SCHEMA.INNODB_TABLESPACES WHERE NAME='test/t1';
+SELECT * FROM t1 LIMIT 10;
+
+--echo ############################################################
+--echo # ALTER TABLESPACE 2 :    Encrypted => Unencrypted         #
+--echo #                         (crash at page 10)               #
+--echo ############################################################
+--echo # Set Unencryption process to crash at page 10
+SET SESSION debug= '+d,alter_encrypt_tablespace_page_10';
+
+--echo # Unencrypt the tablespace. It will cause crash.
+--source include/expect_crash.inc
+--error 0,CR_SERVER_LOST,ER_INTERNAL_ERROR
+ALTER TABLESPACE `test/t1` ENCRYPTION='N';
+--echo # Restart after crash
+--source include/start_mysqld_no_echo.inc
+
+--echo # Wait for Unencryption processing to finish in background thread
+let $wait_condition = SELECT ENCRYPTION = 'N'
+        FROM INFORMATION_SCHEMA.INNODB_TABLESPACES
+        WHERE NAME='test/t1';
+--source include/wait_condition.inc
+
+# Make sure ts file is updated with new records in table
+set global innodb_buf_flush_list_now = 1;
+
+--echo # After restart/recovery, check that Unencryption was roll-forward
+SELECT NAME, ENCRYPTION FROM INFORMATION_SCHEMA.INNODB_TABLESPACES WHERE NAME='test/t1';
+SELECT * FROM t1 LIMIT 10;
+
+--echo #########################################################################
+--echo # RESTART 4 : WITHOUT KEYRING PLUGIN
+--echo #########################################################################
+let $restart_parameters = restart:;
+--source include/restart_mysqld_no_echo.inc
+
+SELECT NAME, ENCRYPTION FROM INFORMATION_SCHEMA.INNODB_TABLESPACES WHERE NAME='test/t1';
+# We should be able to read t1 records even without OKV keyring plugin as it should
```

Comment:
OKV?

Reply:
This test is copy of existing testcase tablespace_encrypt_2. Modified for testing fpt tablespace.
OKV refers to Oracle Keyring Vault plugin